src/lib.rs

```diff
@@ -266,7 +274,15 @@ impl TextEmbedding {
 
     /// Private method to return an instance
     fn new(tokenizer: Tokenizer, session: Session) -> Self {
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (docstrings):** Please update the docstring for function: `TextEmbedding::new`

Reason for update: The function signature and internal logic have changed to include a new field `need_token_type_ids`.

Suggested new docstring:
```
/// Private method to return an instance, determining if `token_type_ids` are needed based on the session inputs.
```